### PR TITLE
assetlibrary-export: optional schedule for export

### PR DESCRIPTION
**Pull request by aaronatbissell**
# Description
Add an optional parameter to the asset library export package to deploy it with a schedule (defaults to no schedule).

## Type of change

- [ ] *Bug fix (non-breaking change which fixes an issue)*
- [x] *New feature (non-breaking change which adds functionality)*
- [ ] *Breaking change (fix or feature that would cause existing functionality to not work as expected)*
- [ ] *Refactor* (existing code being refactored)
- [ ] *This change requires a documentation update*

# Submission Checklist

- [x] Build Verified
- [x] Bundle Verified
- [x] Lint passing
- [x] Unit tests passing
- [ ] Integration tests passing
- [x] Change logs generated

### CODE REVIEW

**Comment by aaronatbissell** [comment]
OK - so the path forward here to get this running on a schedule would be to create a cloudwatch event that triggers the export from within the facade? I'm ok with this approach. Would we need to export the ARN of the state machine then, in order to invoke it?

**Comment by hassankhokhar** [comment]
Yes, cloudwatch event configured in the facade to trigger the state machine. It does seem like the current cfn template doesnt have statemachine arn exported. That would be required for this to work from an external trigger i.e. cloudwatch event. You could also remove the `AssetLibraryExportStateMachineExecutionRole`. I think this role is lingering role which was used for the cloudwatch event config. This can be defined in externally as well.

**Comment by aaronatbissell** [comment]
Great - I'll make those updates to this PR and we'll use it that way

**Comment by aaronatbissell** [comment]
Should be good now - thanks!

File: source/packages/services/assetlibrary-export/infrastructure/cfn-assetlibrary-export.yaml [hunk 1]
```diff
@@ -320,3 +297,9 @@ Outputs:
     Value: !Ref ExportCompletionSNSTopic
     Export:
       Name: !Sub "cdf-assetLibrary-export-${Environment}-exportCompletionSnsTopic"
+
+  ExportStateMachineName:
```
Comment (hassankhokhar) [inline source/packages/services/assetlibrary-export/infrastructure/cfn-assetlibrary-export.yaml:301]: Change `ExportStateMachineName` to `AssetLibraryExportStateMachineArn`

**Review by hassankhokhar — CHANGES_REQUESTED** [review]

**Review by hassankhokhar — APPROVED** [review]

**Pull request merged**